docs/topics/request-response.rst

```diff
@@ -200,11 +214,9 @@ Example::
         self.logger.info("Visited %s", response.url)
 
 In some cases you may be interested in passing arguments to those callback
-functions so you can receive the arguments later, in the second callback. You
-can use the :attr:`Request.meta` attribute for that.
-
-Here's an example of how to pass an item using this mechanism, to populate
-different fields from different pages::
+functions so you can receive the arguments later, in the second callback.
+The following two examples show how to achieve this by using the 
+:attr:`Request.meta` and :attr:`Request.cb_kwargs` attributes respectively::
```

Comment:
I think the documentation should cover only the usage of `Request.cb_kwargs`, and do not mention `meta` but for those cases where it is the only possible way (e.g. parameters for middleware).

Reply:
The original issue talks about "an _alternative_ way to pass arguments to callbacks". Do we want to encourage users to absolutely stop using `Request.meta` to pass data around?

Reply:
I think so.
`cb_kwargs` should be the option for callback data.
`Request.meta` should be the option for other components like _middlewares_.

Reply:
What do you think about mention meta, but saying that in previous Scrapy versions this style was common, but now we encourge you to use callback kwargs for user data, and meta for components like middlewares? People will be reading a lot of code which uses meta, this may help them.

Reply:
Updated, please take another look everyone :eyes: